### ISSUE TITLE
SegFault Compile on AWS Linux

### ISSUE THREAD

**Issue by RichardKav** [adamrehn/ue4-docker#128]
Attempting to compile with the command:

ue4-docker build --cuda=9.2 --pull-prerequisites --exclude=debug --exclude=templates 4.25.0

I get the following segfault.

```
[ec2-user@ip-10-0-1-67 ~]$ screen -r
    Engine crash handling finished; re-raising signal 11 for the default handler. Good bye.
  Took 1204.442652s to run UE4Editor, ExitCode=139
  SafeDeleteFile /home/ue4/Library/Logs/Unreal Engine/LocalBuildLogs/BuildDerivedDataCache/DerivedDataCache-2021.02.05-18.06.38.txt
  SafeCopyFile /home/ue4/UnrealEngine/Engine/Programs/AutomationTool/Saved/DerivedDataCache-2021.02.05-17.46.34.txt /home/ue4/Library/Logs/Unreal Engine/LocalBuildLogs/BuildDerivedDataCache/DerivedDataCache-2021.02.05-18.06.38.txt
  SafeDeleteFile /home/ue4/UnrealEngine/Engine/Programs/AutomationTool/Saved/DerivedDataCache-2021.02.05-17.46.34.txt
  Editor terminated with exit code 139 (segmentation fault) while running DerivedDataCache for /home/ue4/UnrealEngine/Templates/FP_FirstPersonBP/FP_FirstPersonBP.uproject; see log /home/ue4/Library/Logs/Unreal Engine/LocalBuildLogs/BuildDerivedDataCache/DerivedDataCache-2021.02.05-18.06.38.txt
  CommandletException: Editor terminated with exit code 139 (segmentation fault) while running DerivedDataCache for /home/ue4/UnrealEngine/Templates/FP_FirstPersonBP/FP_FirstPersonBP.uproject; see log /home/ue4/Library/Logs/Unreal Engine/LocalBuildLogs/BuildDerivedDataCache/DerivedDataCache-2021.02.05-18.06.38.txt
    at AutomationTool.CommandUtils.RunCommandlet (Tools.DotNETCommon.FileReference ProjectName, System.String UE4Exe, System.String Commandlet, System.String Parameters, System.String& DestLogFile) [0x0076d] in <70cb8c954e26476ca043382399d367f7>:0
    at AutomationTool.CommandUtils.RunCommandlet (Tools.DotNETCommon.FileReference ProjectName, System.String UE4Exe, System.String Commandlet, System.String Parameters) [0x00001] in <70cb8c954e26476ca043382399d367f7>:0
    at AutomationTool.CommandUtils.DDCCommandlet (Tools.DotNETCommon.FileReference ProjectName, System.String UE4Exe, System.String[] Maps, System.String TargetPlatform, System.String Parameters) [0x00046] in <70cb8c954e26476ca043382399d367f7>:0
    at BuildDerivedDataCache.ExecuteBuild () [0x00279] in <845bbc68e9354c37a0f45fa2db896d3e>:0
    at AutomationTool.BuildCommand.Execute () [0x00001] in <70cb8c954e26476ca043382399d367f7>:0
    at AutomationTool.Automation.Execute (System.Collections.Generic.List`1[T] CommandsToExecute, System.Collections.Generic.Dictionary`2[TKey,TValue] Commands) [0x0007c] in <70cb8c954e26476ca043382399d367f7>:0
    at AutomationTool.Automation.Process (System.String[] Arguments, UnrealBuildTool.StartupTraceListener StartupListener) [0x00284] in <70cb8c954e26476ca043382399d367f7>:0
    at AutomationTool.Program.MainProc (System.String[] Arguments, UnrealBuildTool.StartupTraceListener StartupListener) [0x00001] in <9240e6d070624b0da722632255a242d2>:0
    at AutomationTool.Program+<>c__DisplayClass1_0.<Main>b__2 () [0x00000] in <9240e6d070624b0da722632255a242d2>:0
    at AutomationTool.InternalUtils.RunSingleInstance (System.Func`1[TResult] Main) [0x000a0] in <70cb8c954e26476ca043382399d367f7>:0
    at AutomationTool.Program.Main (System.String[] Arguments) [0x0020a] in <9240e6d070624b0da722632255a242d2>:0
  AutomationTool exiting with ExitCode=1 (Error_Unknown)
Took 1208.678899s to run mono, ExitCode=1
AutomationTool exiting with ExitCode=1 (Error_Unknown)
RunUAT ERROR: AutomationTool was unable to run successfully.
The command '/bin/sh -c ./Engine/Build/BatchFiles/RunUAT.sh BuildGraph -target="Make Installed Build Linux" -script=Engine/Build/InstalledEngineBuild.xml -set:HostPlatformOnly=true -set:WithDDC=$BUILD_DDC &&    rm -R -f /home/ue4/UnrealEngine/LocalBuilds/InstalledDDC &&        python3 /tmp/exclude-components.py /home/ue4/UnrealEngine/LocalBuilds/Engine/Linux $EXCLUDE_DEBUG $EXCLUDE_TEMPLATES && echo '' && echo 'RUN directive complete. Docker will now commit the filesystem layer to disk.' && echo 'Note that for large filesystem layers this can take quite some time.' && echo 'Performing filesystem layer commit...' && echo ''' returned a non-zero code: 1
[ue4-docker build] Error: failed to build image "adamrehn/ue4-minimal:4.25.0-cudagl9.2".
[ec2-user@ip-10-0-1-67 ~]$

```
It is using a "AWS Deep Learning Base AMI (Amazon Linux 2)" based image, with a g4dn.xlarge instance.
The output of the `ue4-docker info` command is:

```
ue4-docker info
ue4-docker version:         0.0.77 (latest available version is 0.0.77)
Operating system:           Linux (Amazon Linux AMI 2018.03, 4.14.200-116.320.amzn1.x86_64)
Docker daemon version:      19.03.6-ce
NVIDIA Docker supported:    Yes
Maximum image size:         No limit detected
Available disk space:       659.19 GiB
Total system memory:        15.38 GiB physical, 0 bytes virtual
Number of processors:       2 physical, 4 logical
```

**Comment by adamrehn** [comment]
The .0 release of pretty much every version of the Unreal Engine is broken in some way, and more often than not the bugs break the build process under Linux. Can you please try building Unreal Engine version 4.25.4 instead and see if that resolves your issue?

**Comment by RichardKav** [comment]
Avoiding the .0 release seems to have got further, but is now having issues with python packages.

```
[ue4-docker build] Building image "adamrehn/ue4-full:4.25.4-cudagl10.0"...
Sending build context to Docker daemon  9.216kB
Step 1/35 : ARG NAMESPACE
Step 2/35 : ARG TAG
Step 3/35 : ARG PREREQS_TAG
Step 4/35 : ARG UE4CLI_VERSION="ue4cli>=0.0.45"
Step 5/35 : ARG CONAN_UE4CLI_VERSION="conan-ue4cli>=0.0.27"
Step 6/35 : FROM ${NAMESPACE}/ue4-source:${TAG}-${PREREQS_TAG} AS builder
 ---> 58ce9d62c286
Step 7/35 : ARG UE4CLI_VERSION
 ---> Using cache
 ---> 142090724a02
Step 8/35 : ARG CONAN_UE4CLI_VERSION
 ---> Using cache
 ---> 079cfc5626cc
Step 9/35 : USER root
 ---> Using cache
 ---> 8a82ec94b706
Step 10/35 : RUN pip3 install setuptools wheel && echo '' && echo 'RUN directive complete. Docker will now commit the filesystem layer to disk.' && echo 'Note that for large filesystem layers this can take quite some time.' && echo 'Performing filesystem layer commit...' && echo ''
 ---> Using cache
 ---> 25252a2c3e20
Step 11/35 : RUN pip3 install "$UE4CLI_VERSION" "$CONAN_UE4CLI_VERSION" && echo '' && echo 'RUN directive complete. Docker will now commit the filesystem layer to disk.' && echo 'Note that for large filesystem layers this can take quite some time.' && echo 'Performing filesystem layer commit...' && echo ''
 ---> Running in fb3c8d39fe27
Collecting ue4cli>=0.0.45
  Downloading https://files.pythonhosted.org/packages/0b/da/856716daba2bd306b7c764a03b14da6e95ed6c3da92caa2e0e6d3876f801/ue4cli-0.0.51-py3-none-any.whl
Collecting conan-ue4cli>=0.0.27
  Downloading https://files.pythonhosted.org/packages/ff/5a/1c2833507d79ac62c1f96ff78aff0c079611177de142e763604b50391cdf/conan_ue4cli-0.0.35-py3-none-any.whl (47kB)
Collecting shellescape (from ue4cli>=0.0.45)
  Downloading https://files.pythonhosted.org/packages/d0/f4/0081137fceff5779cd4205c1e96657e41cc2d2d56c940dc8eeb6111780f7/shellescape-3.8.1-py2.py3-none-any.whl
Collecting twine>=1.11.0 (from ue4cli>=0.0.45)
  Downloading https://files.pythonhosted.org/packages/eb/d7/a46b7eb07b491a9295c3581f86ad68f97df83e000dc9a4c6e3c710c432dd/twine-3.3.0-py3-none-any.whl
Requirement already satisfied: setuptools>=38.6.0 in /usr/local/lib/python3.6/dist-packages (from ue4cli>=0.0.45)
Requirement already satisfied: wheel>=0.31.0 in /usr/local/lib/python3.6/dist-packages (from ue4cli>=0.0.45)
Collecting conan>=1.7.4 (from conan-ue4cli>=0.0.27)
  Downloading https://files.pythonhosted.org/packages/ce/83/9d14a3445882741ef8219fff5b7eabf09b2a4223c32b1e5352686474d6ba/conan-1.33.1.tar.gz (639kB)
Collecting tqdm>=4.14 (from twine>=1.11.0->ue4cli>=0.0.45)
  Downloading https://files.pythonhosted.org/packages/d9/13/f3f815bb73804a8af9cfbb6f084821c037109108885f46131045e8cf044e/tqdm-4.57.0-py2.py3-none-any.whl (72kB)
Collecting rfc3986>=1.4.0 (from twine>=1.11.0->ue4cli>=0.0.45)
  Downloading https://files.pythonhosted.org/packages/78/be/7b8b99fd74ff5684225f50dd0e865393d2265656ef3b4ba9eaaaffe622b8/rfc3986-1.4.0-py2.py3-none-any.whl
Collecting importlib-metadata; python_version < "3.8" (from twine>=1.11.0->ue4cli>=0.0.45)
  Downloading https://files.pythonhosted.org/packages/3a/0d/af9e3dce6524461fa1e8327449f392edac8a3d880b4c91ce3e2d25450d03/importlib_metadata-3.7.0-py3-none-any.whl
Collecting pkginfo>=1.4.2 (from twine>=1.11.0->ue4cli>=0.0.45)
  Downloading https://files.pythonhosted.org/packages/b0/9a/d78e7c299eb5659bc3a036e5a968a399c62bfe0b2aa18baf7d13f43373ba/pkginfo-1.7.0-py2.py3-none-any.whl
Collecting requests>=2.20 (from twine>=1.11.0->ue4cli>=0.0.45)
  Downloading https://files.pythonhosted.org/packages/29/c1/24814557f1d22c56d50280771a17307e6bf87b70727d975fd6b2ce6b014a/requests-2.25.1-py2.py3-none-any.whl (61kB)
Collecting requests-toolbelt!=0.9.0,>=0.8.0 (from twine>=1.11.0->ue4cli>=0.0.45)
  Downloading https://files.pythonhosted.org/packages/60/ef/7681134338fc097acef8d9b2f8abe0458e4d87559c689a8c306d0957ece5/requests_toolbelt-0.9.1-py2.py3-none-any.whl (54kB)
Collecting keyring>=15.1 (from twine>=1.11.0->ue4cli>=0.0.45)
  Downloading https://files.pythonhosted.org/packages/96/17/86e1e5b50dc14e5984a661dde2ed3b7f05efe1fdee32145a77708cf06904/keyring-22.0.1-py3-none-any.whl
Collecting readme-renderer>=21.0 (from twine>=1.11.0->ue4cli>=0.0.45)
  Downloading https://files.pythonhosted.org/packages/39/a5/459adfa22ea237f6e8d0fa95ad29d7369579a5eec26f016ab34bb7f8359c/readme_renderer-29.0-py2.py3-none-any.whl
Collecting colorama>=0.4.3 (from twine>=1.11.0->ue4cli>=0.0.45)
  Downloading https://files.pythonhosted.org/packages/44/98/5b86278fbbf250d239ae0ecb724f8572af1c91f4a11edf4d36a206189440/colorama-0.4.4-py2.py3-none-any.whl
Collecting PyJWT<2.0.0,>=1.4.0 (from conan>=1.7.4->conan-ue4cli>=0.0.27)
  Downloading https://files.pythonhosted.org/packages/87/8b/6a9f14b5f781697e51259d81657e6048fd31a113229cf346880bb7545565/PyJWT-1.7.1-py2.py3-none-any.whl
Collecting urllib3!=1.25.4,!=1.25.5,<1.26 (from conan>=1.7.4->conan-ue4cli>=0.0.27)
  Downloading https://files.pythonhosted.org/packages/56/aa/4ef5aa67a9a62505db124a5cb5262332d1d4153462eb8fd89c9fa41e5d92/urllib3-1.25.11-py2.py3-none-any.whl (127kB)
Collecting PyYAML<6.0,>=3.11 (from conan>=1.7.4->conan-ue4cli>=0.0.27)
  Downloading https://files.pythonhosted.org/packages/7a/5b/bc0b5ab38247bba158504a410112b6c03f153c652734ece1849749e5f518/PyYAML-5.4.1-cp36-cp36m-manylinux1_x86_64.whl (640kB)
Collecting patch-ng<1.18,>=1.17.4 (from conan>=1.7.4->conan-ue4cli>=0.0.27)
  Downloading https://files.pythonhosted.org/packages/c1/b2/ad3cd464101435fdf642d20e0e5e782b4edaef1affdf2adfc5c75660225b/patch-ng-1.17.4.tar.gz
Collecting fasteners>=0.14.1 (from conan>=1.7.4->conan-ue4cli>=0.0.27)
  Downloading https://files.pythonhosted.org/packages/78/20/c862d765287e9e8b29f826749ebae8775bdca50b2cb2ca079346d5fbfd76/fasteners-0.16-py2.py3-none-any.whl
Collecting six<=1.15.0,>=1.10.0 (from conan>=1.7.4->conan-ue4cli>=0.0.27)
  Downloading https://files.pythonhosted.org/packages/ee/ff/48bde5c0f013094d729fe4b0316ba2a24774b3ff1c52d924a8a4cb04078a/six-1.15.0-py2.py3-none-any.whl
Collecting node-semver==0.6.1 (from conan>=1.7.4->conan-ue4cli>=0.0.27)
  Downloading https://files.pythonhosted.org/packages/08/51/6cf3a2b18ca35cbe4ad3c7538a7c3dc0cb24e71629fb16e729c137d06432/node_semver-0.6.1-py3-none-any.whl
Collecting distro<=1.5.0,>=1.0.2 (from conan>=1.7.4->conan-ue4cli>=0.0.27)
  Downloading https://files.pythonhosted.org/packages/25/b7/b3c4270a11414cb22c6352ebc7a83aaa3712043be29daa05018fd5a5c956/distro-1.5.0-py2.py3-none-any.whl
Collecting future<0.19.0,>=0.16.0 (from conan>=1.7.4->conan-ue4cli>=0.0.27)
  Downloading https://files.pythonhosted.org/packages/45/0b/38b06fd9b92dc2b68d58b75f900e97884c45bedd2ff83203d933cf5851c9/future-0.18.2.tar.gz (829kB)
Collecting pygments<3.0,>=2.0 (from conan>=1.7.4->conan-ue4cli>=0.0.27)
  Downloading https://files.pythonhosted.org/packages/d4/ef/881b14d3ad4d6a86384d63bfbb87205161e4fc46420526380ec209a12e17/Pygments-2.8.0-py3-none-any.whl (983kB)
Collecting deprecation<2.1,>=2.0 (from conan>=1.7.4->conan-ue4cli>=0.0.27)
  Downloading https://files.pythonhosted.org/packages/b9/2a/d5084a8781398cea745c01237b95d9762c382697c63760a95cc6a814ad3a/deprecation-2.0.7-py2.py3-none-any.whl
Collecting Jinja2<3,>=2.9 (from conan>=1.7.4->conan-ue4cli>=0.0.27)
  Downloading https://files.pythonhosted.org/packages/7e/c2/1eece8c95ddbc9b1aeb64f5783a9e07a286de42191b7204d67b7496ddf35/Jinja2-2.11.3-py2.py3-none-any.whl (125kB)
Collecting python-dateutil<3,>=2.7.0 (from conan>=1.7.4->conan-ue4cli>=0.0.27)
  Downloading https://files.pythonhosted.org/packages/d4/70/d60450c3dd48ef87586924207ae8907090de0b306af2bce5d134d78615cb/python_dateutil-2.8.1-py2.py3-none-any.whl (227kB)
Collecting bottle<0.13,>=0.12.8 (from conan>=1.7.4->conan-ue4cli>=0.0.27)
  Downloading https://files.pythonhosted.org/packages/bf/44/aeafdd6ca05a8e1c3f91eeeb272a202d5cb1b3b23730a5ca686a81c48d24/bottle-0.12.19-py3-none-any.whl (89kB)
Collecting pluginbase>=0.5 (from conan>=1.7.4->conan-ue4cli>=0.0.27)
  Downloading https://files.pythonhosted.org/packages/3d/3c/fe974b4f835f83cc46966e04051f8708b7535bac28fbc0dcca1ee0c237b8/pluginbase-1.0.0.tar.gz (41kB)
Collecting zipp>=0.5 (from importlib-metadata; python_version < "3.8"->twine>=1.11.0->ue4cli>=0.0.45)
  Downloading https://files.pythonhosted.org/packages/41/ad/6a4f1a124b325618a7fb758b885b68ff7b058eec47d9220a12ab38d90b1f/zipp-3.4.0-py3-none-any.whl
Collecting typing-extensions>=3.6.4; python_version < "3.8" (from importlib-metadata; python_version < "3.8"->twine>=1.11.0->ue4cli>=0.0.45)
  Downloading https://files.pythonhosted.org/packages/60/7a/e881b5abb54db0e6e671ab088d079c57ce54e8a01a3ca443f561ccadb37e/typing_extensions-3.7.4.3-py3-none-any.whl
Collecting certifi>=2017.4.17 (from requests>=2.20->twine>=1.11.0->ue4cli>=0.0.45)
  Downloading https://files.pythonhosted.org/packages/5e/a0/5f06e1e1d463903cf0c0eebeb751791119ed7a4b3737fdc9a77f1cdfb51f/certifi-2020.12.5-py2.py3-none-any.whl (147kB)
Collecting chardet<5,>=3.0.2 (from requests>=2.20->twine>=1.11.0->ue4cli>=0.0.45)
  Downloading https://files.pythonhosted.org/packages/19/c7/fa589626997dd07bd87d9269342ccb74b1720384a4d739a1872bd84fbe68/chardet-4.0.0-py2.py3-none-any.whl (178kB)
Collecting idna<3,>=2.5 (from requests>=2.20->twine>=1.11.0->ue4cli>=0.0.45)
  Downloading https://files.pythonhosted.org/packages/a2/38/928ddce2273eaa564f6f50de919327bf3a00f091b5baba8dfa9460f3a8a8/idna-2.10-py2.py3-none-any.whl (58kB)
Collecting jeepney>=0.4.2; sys_platform == "linux" (from keyring>=15.1->twine>=1.11.0->ue4cli>=0.0.45)
  Downloading https://files.pythonhosted.org/packages/51/b0/a6ea72741aaac3f37fb96d195e4ee576a103c4c04e279bc6b446a70960e1/jeepney-0.6.0-py3-none-any.whl (45kB)
Collecting SecretStorage>=3.2; sys_platform == "linux" (from keyring>=15.1->twine>=1.11.0->ue4cli>=0.0.45)
  Downloading https://files.pythonhosted.org/packages/d9/1e/29cd69fdac7391aa51510dfd42aa70b4e6a826c8cd019ee2a8ab9ec0777f/SecretStorage-3.3.1-py3-none-any.whl
Collecting bleach>=2.1.0 (from readme-renderer>=21.0->twine>=1.11.0->ue4cli>=0.0.45)
  Downloading https://files.pythonhosted.org/packages/f0/46/2bbd92086a4c6f051214cb48df6d9132b5f32c5e881d3f4991b16ec7e499/bleach-3.3.0-py2.py3-none-any.whl (283kB)
Collecting docutils>=0.13.1 (from readme-renderer>=21.0->twine>=1.11.0->ue4cli>=0.0.45)
  Downloading https://files.pythonhosted.org/packages/81/44/8a15e45ffa96e6cf82956dd8d7af9e666357e16b0d93b253903475ee947f/docutils-0.16-py2.py3-none-any.whl (548kB)
Collecting packaging (from deprecation<2.1,>=2.0->conan>=1.7.4->conan-ue4cli>=0.0.27)
  Downloading https://files.pythonhosted.org/packages/3e/89/7ea760b4daa42653ece2380531c90f64788d979110a2ab51049d92f408af/packaging-20.9-py2.py3-none-any.whl (40kB)
Collecting MarkupSafe>=0.23 (from Jinja2<3,>=2.9->conan>=1.7.4->conan-ue4cli>=0.0.27)
  Downloading https://files.pythonhosted.org/packages/b2/5f/23e0023be6bb885d00ffbefad2942bc51a620328ee910f64abe5a8d18dd1/MarkupSafe-1.1.1-cp36-cp36m-manylinux1_x86_64.whl
Collecting cryptography>=2.0 (from SecretStorage>=3.2; sys_platform == "linux"->keyring>=15.1->twine>=1.11.0->ue4cli>=0.0.45)
  Downloading https://files.pythonhosted.org/packages/fa/2d/2154d8cb773064570f48ec0b60258a4522490fcb115a6c7c9423482ca993/cryptography-3.4.6.tar.gz (546kB)
    Complete output from command python setup.py egg_info:

            =============================DEBUG ASSISTANCE==========================
            If you are seeing an error here please try the following to
            successfully install cryptography:

            Upgrade to the latest pip and try again. This will fix errors for most
            users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip
            =============================DEBUG ASSISTANCE==========================

    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-upojuozb/cryptography/setup.py", line 14, in <module>
        from setuptools_rust import RustExtension
    ModuleNotFoundError: No module named 'setuptools_rust'

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-upojuozb/cryptography/
The command '/bin/sh -c pip3 install "$UE4CLI_VERSION" "$CONAN_UE4CLI_VERSION" && echo '' && echo 'RUN directive complete. Docker will now commit the filesystem layer to disk.' && echo 'Note that for large filesystem layers this can take quite some time.' && echo 'Performing filesystem layer commit...' && echo ''' returned a non-zero code: 1
[ue4-docker build] Error: failed to build image "adamrehn/ue4-full:4.25.4-cudagl10.0".
[ec2-user@ip-10-0-1-67 ~]$
[ec2-user@ip-10-0-1-67 ~]$ pip3 --version
pip 21.0.1 from /usr/local/lib/python3.6/site-packages/pip (python 3.6)
[ec2-user@ip-10-0-1-67 ~]$
```

**Comment by adamrehn** [comment]
That error was fixed in #129, which is live in ue4-docker version 0.0.78 and newer. Can you please update to the latest version of ue4-docker and try again?

**Comment by RichardKav** [comment]
On upgrading the version of ue4-docker it has now successfully built the images. Thanks.
